### PR TITLE
chore: remove swap and buy buttons from home and assets details screen except for US countries

### PR DESCRIPTION
**Pull request by dangelovska**

### CODE REVIEW

File: brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 1]
```diff
@@ -326,6 +324,30 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber {
         tabBar.items = buttons
     }
     
+    private func updateToolbar() {
+        UserManager.shared.refresh { [weak self] _ in
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:328]: Pull to refresh on home screen already reloads the user, so you are reloading the user twice here

File: brd-ios/breadwallet/src/Views/AssetDetailsFooterView.swift [hunk 2]
```diff
@@ -75,8 +85,7 @@ class AssetDetailsFooterView: UIView, Subscriber {
         buttonsView.snp.makeConstraints { make in
             make.top.equalToSuperview().offset(Margins.medium.rawValue)
             make.height.equalTo(BottomBarItem.defaultheight)
-            make.leading.equalToSuperview().offset(Margins.huge.rawValue)
-            make.trailing.equalToSuperview().offset(-Margins.huge.rawValue)
+            make.centerX.equalToSuperview()
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Views/AssetDetailsFooterView.swift:88]: why did you change this? if i'm not mistaken this pushes both buttons near the center, instead of being equally spaced out on the bottom bar.

File: brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 3]
```diff
@@ -326,6 +324,30 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber {
...
+            let hasSwapBuyAccess = UserManager.shared.profile?.hasSwapBuyAccess ?? false
+            
+            if hasSwapBuyAccess && self?.tabBarButtons.count != 5 {
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:331]: i don't think we need to check for number of tabbarbuttons in any of the cases

File: brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 4]
```diff
@@ -324,6 +324,42 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber {
         }
         
         tabBar.items = buttons
+        tabBarItemsNumber = tabBarButtons.count
+    }
+    
+    private func updateToolbar() {
+        guard let hasSwapBuyAccess = UserManager.shared.profile?.hasSwapBuyAccess else {
+            UserManager.shared.refresh { [weak self] _ in
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:332]: When you do pull to refresh on home screen the user profile is already refreshed, we don't need to refresh it again here

Comment (dangelovska) [reply brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:332]: I need this on first launch on the app, the tab bar need to be shown correctly

File: brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 5]
```diff
@@ -324,6 +324,42 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber {
         }
         
         tabBar.items = buttons
+        tabBarItemsNumber = tabBarButtons.count
+    }
+    
+    private func updateToolbar() {
+        guard let hasSwapBuyAccess = UserManager.shared.profile?.hasSwapBuyAccess else {
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:331]: you added local var hasSwapBuyAccess to HomeScreenVC above and you are not using it. When we do pull to refresh simply assigned the value of UserManager.shared.profile?.hasSwapBuyAccess to that local variable. Then add didSet logic to the local variable and move the logic to setting 3 or 5 buttons to there, and you can add a guard at the start that checks if new value is same as previous value and return in that case, so nothing gets executed.

Comment (dangelovska) [reply brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:331]: we don't need to update the tabBar on pullToRefresh option, with my solution it is checking for update on every login, logout.. every time when we open HS and we need to update the tabBar

Comment (dangelovska) [reply brd-ios/breadwallet/src/ViewControllers/HomeScreenViewController.swift:331]: and I am refreshing the profile only on first launch, because we don't have profile data yet

**Review by dgacevic — APPROVED** [review]
